src/storage-controller/src/lib.rs

```diff
@@ -2529,6 +2583,15 @@ where
                 collection
             );
 
+            // Keep track of the timestamp of the latest row per id
+            let timestamp = occurred_at.unwrap_timestamptz();
+            match latest_row_per_id.get(&id) {
+                Some(existing) if &existing.0 > &timestamp => {}
+                _ => {
+                    latest_row_per_id.insert(id, (timestamp, status_row.clone()));
+                }
+            }
```

Comment:
```suggestion
            let timestamp = occurred_at.unwrap_timestamptz();
            latest_row_per_id.entry(id)
            	.and_modify(
            		|existing| if &existing.0 < &timestamp { *existing = (...) }
            	.or_insert(...)
 ```

you may prefer this, but its your call!

Reply:
The functional approach is a little more fussy because and a few more LOC. Agree that this is something that should be prettier to do.